som/learning_rate.go

```diff
@@ -0,0 +1,28 @@
+package som
+
+import "math"
+
+const SmallestRate = 0.01
+
+// LearningRate is a decay function with two available strategies: exponential and linear.
+// Both strategies return the learningRate0 for iteration 0 and SmallestRate
+// for iteration == totalIterations-1
+func LearningRate(iteration, totalIterations int, strategy string, learningRate0 float64) float64 {
```

Comment:
Right, reading an article on errors in go, I understand what you say.  Though part of me is really reluctant to making code fool proof.  Can we just check the radius0 is not negative (that would break the code because of `math.Log`)?  If someone wants to pass negative iteration number, well, it's their problem I think.